boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -96,6 +96,12 @@ NM = $(CROSSDEV)nm
 OBJCOPY = $(CROSSDEV)objcopy
 OBJDUMP = $(CROSSDEV)objdump
 
+# clang based native compiler doesn't need opjcopy to build simulation.
+# Note: In fact objcopy is not part of MacOS native compiler suite.
+ifeq ($(CONFIG_HOST_TOOLCHAIN_CLANGL),y)  
+  OBJCOPY = :
```

Comment:
from this page:
https://llvm.org/docs/CommandGuide/index.html
llvm-ar, llvm-strip, llvm-nm, llvm-objdump, llvm-objcopy

Reply:
On my MacOS/M1/Monterey on llvm-g++  and llvm-gcc are available. Additionally, any 'llvm-' tool would be not used unless you link them or specify the CROSSDEV prefix as defined in boards/sim/sim/sim/scripts/Make.def.

Reply:
let's directly skip objcopy in arch/sim/Makefile?